### ISSUE TITLE
ClickHouse: tuple element access .N on a parenthesised / subscript / literal expression is unparsable

### ISSUE THREAD

**Issue by friedrich123456789** [sqlfluff/sqlfluff#8032]
### What happened

Tuple element access by dot-index works on a bare identifier (`x.1`, `x.1.2`
parse) but fails whenever the left-hand side ends in `)` or `]` — i.e. a
function call, a subscript, or a tuple literal.

Follow-up to #3816 (tuple access, fixed for the identifier case) — the
expression-prefixed cases remain.

### Expected behaviour

`<expr>.N` tuple-element access parses whenever `<expr>` is a tuple-valued
expression, including function calls, subscripts and tuple literals.

### Observed behaviour

```
SELECT f(x).2 FROM t    ->  L1:P12 PRS: Found unparsable section: '.2'
SELECT arr[1].2 FROM t  ->  L1:P14 PRS: Found unparsable section: '.2'
SELECT (a, b).1 FROM t  ->  L1:P14 PRS: Found unparsable section: '.1'
```

For contrast, these already parse: `SELECT x.1 FROM t`, `SELECT x.1.2 FROM t`.

### How to reproduce

```sql
SELECT f(x).2 FROM t
SELECT arr[1].2 FROM t
SELECT (a, b).1 FROM t
```

```bash
sqlfluff parse --dialect clickhouse -
```

Valid, verified on ClickHouse 26.4.2 (`(tuple(1,2)).2`, `[tuple(1,2)][1].2`,
`(a, b).1` all accepted).
Docs: https://clickhouse.com/docs/sql-reference/data-types/tuple

### Dialect

clickhouse

### Version

sqlfluff 4.2.2 (latest release), Python 3.12; also applies to `main`.

### Configuration

None — default rules, `--dialect clickhouse`.
